{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "14fe0cf5-3a18-4b8a-8479-65a073bf4762",
   "metadata": {},
   "source": [
    "## Read Nirvana JSON output by lines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "0afb71d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"header object:  {\\\"annotator\\\":\\\"Nirvana 3.16.1\\\",\\\"creationTime\\\":\\\"2022-05-21 10:27:54\\\",\\\"genomeAssembly\\\":\\\"GRCh38\\\",\\\"schemaVersion\\\":6,\\\"dataVersion\\\":\\\"91.27.63\\\",\\\"dataSources\\\":[{\\\"name\\\":\\\"VEP\\\",\\\"version\\\":\\\"91\\\",\\\"description\\\":\\\"BothRefSeqAndEnsembl\\\",\\\"releaseDate\\\":\\\"2017-12-18\\\"},{\\\"name\\\":\\\"MultiZ100Way\\\",\\\"version\\\":\\\"20171006\\\",\\\"description\\\":\\\"Amino acid conservation scores calculated from MultiZ100Way alignments from UCSC.\\\",\\\"releaseDate\\\":\\\"2017-10-06\\\"},{\\\"name\\\":\\\"ClinVar\\\",\\\"version\\\":\\\"20211202\\\",\\\"description\\\":\\\"A freely accessible, public archive of reports of the relationships among human variations and phenotypes, with supporting evidence\\\",\\\"releaseDate\\\":\\\"2021-12-02\\\"},{\\\"name\\\":\\\"dbSNP\\\",\\\"version\\\":\\\"154\\\",\\\"description\\\":\\\"Identifiers for observed variants\\\",\\\"releaseDate\\\":\\\"2020-05-01\\\"},{\\\"name\\\":\\\"dbSNP\\\",\\\"version\\\":\\\"151\\\",\\\"description\\\":\\\"Identifiers for observed variants\\\",\\\"releaseDate\\\":\\\"2018-04-18\\\"},{\\\"name\\\":\\\"gnomAD\\\",\\\"version\\\":\\\"3.1\\\",\\\"description\\\":\\\"Allele frequencies from Genome Aggregation Database (gnomAD)\\\",\\\"releaseDate\\\":\\\"2020-10-29\\\"},{\\\"name\\\":\\\"MITOMAP\\\",\\\"version\\\":\\\"20200819\\\",\\\"description\\\":\\\"Small variants in the MITOMAP human mitochondrial genome database\\\",\\\"releaseDate\\\":\\\"2020-08-19\\\"},{\\\"name\\\":\\\"1000 Genomes Project\\\",\\\"version\\\":\\\"Phase 3 v3plus\\\",\\\"description\\\":\\\"A public catalogue of human variation and genotype data\\\",\\\"releaseDate\\\":\\\"2013-05-27\\\"},{\\\"name\\\":\\\"PrimateAI\\\",\\\"version\\\":\\\"0.2\\\",\\\"description\\\":\\\"PrimateAI percentile scores.\\\",\\\"releaseDate\\\":\\\"2018-11-07\\\"},{\\\"name\\\":\\\"REVEL\\\",\\\"version\\\":\\\"20200205\\\",\\\"description\\\":\\\"Pathogenicity scores of missense variants predicted by REVEL\\\",\\\"releaseDate\\\":\\\"2020-02-05\\\"},{\\\"name\\\":\\\"SpliceAI\\\",\\\"version\\\":\\\"1.3\\\",\\\"description\\\":\\\"SpliceAI scores, distances, etc.\\\",\\\"releaseDate\\\":\\\"2019-10-04\\\"},{\\\"name\\\":\\\"TOPMed\\\",\\\"version\\\":\\\"freeze_5\\\",\\\"description\\\":\\\"Allele frequencies from TOPMed data\\\",\\\"releaseDate\\\":\\\"2017-08-28\\\"},{\\\"name\\\":\\\"ClinGen\\\",\\\"version\\\":\\\"20160414\\\",\\\"releaseDate\\\":\\\"2016-04-14\\\"},{\\\"name\\\":\\\"ClinGen Dosage Sensitivity Map\\\",\\\"version\\\":\\\"20211201\\\",\\\"description\\\":\\\"Dosage sensitivity map from ClinGen (dbVar)\\\",\\\"releaseDate\\\":\\\"2021-12-01\\\"},{\\\"name\\\":\\\"MITOMAP_SV\\\",\\\"version\\\":\\\"20200819\\\",\\\"description\\\":\\\"Large structural variants in the MITOMAP human mitochondrial genome database\\\",\\\"releaseDate\\\":\\\"2020-08-19\\\"},{\\\"name\\\":\\\"1000 Genomes Project (SV)\\\",\\\"version\\\":\\\"Phase 3 v5a\\\",\\\"description\\\":\\\"A public catalogue of human variation and genotype data\\\",\\\"releaseDate\\\":\\\"2013-05-27\\\"},{\\\"name\\\":\\\"COSMIC gene fusions\\\",\\\"version\\\":\\\"94\\\",\\\"description\\\":\\\"manually curated somatic gene fusions\\\",\\\"releaseDate\\\":\\\"2021-05-28\\\"},{\\\"name\\\":\\\"FusionCatcher\\\",\\\"version\\\":\\\"1.33\\\",\\\"description\\\":\\\"known germline and somatic gene fusions\\\",\\\"releaseDate\\\":\\\"2020-12-22\\\"},{\\\"name\\\":\\\"ClinGen disease validity curations\\\",\\\"version\\\":\\\"20211201\\\",\\\"description\\\":\\\"Disease validity curations from ClinGen (dbVar)\\\",\\\"releaseDate\\\":\\\"2021-12-01\\\"},{\\\"name\\\":\\\"gnomAD_gene_scores\\\",\\\"version\\\":\\\"2.1\\\",\\\"description\\\":\\\"Gene Loss of function scores from Genome Aggregation Database (gnomAD)\\\",\\\"releaseDate\\\":\\\"2018-10-17\\\"},{\\\"name\\\":\\\"OMIM\\\",\\\"version\\\":\\\"20211202\\\",\\\"description\\\":\\\"An Online Catalog of Human Genes and Genetic Disorders\\\",\\\"releaseDate\\\":\\\"2021-12-02\\\"},{\\\"name\\\":\\\"phyloP\\\",\\\"version\\\":\\\"hg38\\\",\\\"description\\\":\\\"20 way conservation score between humans and 19 other vertebrates\\\",\\\"releaseDate\\\":\\\"2015-04-17\\\"},{\\\"name\\\":\\\"gnomAD_LCR\\\",\\\"version\\\":\\\"2.1\\\",\\\"description\\\":\\\"Low complexity regions from gnomAD.\\\",\\\"releaseDate\\\":\\\"2019-04-10\\\"},{\\\"name\\\":\\\"MitochondrialHeteroplasmy\\\",\\\"version\\\":\\\"20180410\\\",\\\"description\\\":\\\"Variant read frequency percentiles for the Mitochondrial reference\\\",\\\"releaseDate\\\":\\\"2020-05-21\\\"}],\\\"samples\\\":[\\\"S1P\\\"]}\"\n",
      "[1] \"number of positions:  1941\"\n",
      "[1] \"number of genes:  27\"\n"
     ]
    }
   ],
   "source": [
    "file = gzfile('S1P_dragen_subset.json.gz', 'rt')\n",
    "\n",
    "header = ''\n",
    "positions = c()\n",
    "genes = c()\n",
    "is_header_line = TRUE\n",
    "is_position_line = FALSE\n",
    "is_gene_line = FALSE\n",
    "gene_section_line = '],\"genes\":['\n",
    "end_line = ']}'\n",
    "position_count = 0\n",
    "gene_count = 0\n",
    "while ( TRUE ) {\n",
    "    line = readLines(file, n = 1)\n",
    "    trim_line = trimws(line)\n",
    "    if ( is_header_line ) {\n",
    "        ## only keep the \"header\" field content from the line\n",
    "        header = substr(trim_line, 11, nchar(trim_line)-14)\n",
    "        is_header_line = FALSE\n",
    "        is_position_line = TRUE\n",
    "        next\n",
    "    }\n",
    "    if ( trim_line == gene_section_line ) {\n",
    "        is_gene_line = TRUE\n",
    "        is_position_line = FALSE\n",
    "        next\n",
    "    }\n",
    "    else if ( trim_line == end_line ) {\n",
    "        break\n",
    "    }\n",
    "    else {\n",
    "        if ( is_position_line ) {\n",
    "            ## remove the trailing ',' if there is\n",
    "            positions = c(positions, sub(\",$\", \"\", trim_line))\n",
    "            position_count = position_count + 1\n",
    "        }\n",
    "        if ( is_gene_line ) {\n",
    "            ## remove the trailing ',' if there is\n",
    "            genes = c(genes, sub(\",$\", \"\", trim_line))\n",
    "            gene_count = gene_count + 1\n",
    "        }\n",
    "    }\n",
    "}\n",
    "close(file)\n",
    "print(paste('header object: ', header))\n",
    "print(paste('number of positions: ', position_count))\n",
    "print(paste('number of genes: ', gene_count))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41a160e1-bb63-4192-abd5-4f474c6e7399",
   "metadata": {},
   "source": [
    "## Retrieve variants under a gnomAD allele frequency threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9a39621d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 90  2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 20 × 2 of type chr</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>variant_id</th><th scope=col>gnomAD_allele_freq</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>1-28198-A-T </td><td>0      </td></tr>\n",
       "\t<tr><td>1-46402-C-CT</td><td>0      </td></tr>\n",
       "\t<tr><td>1-55119-C-G </td><td>2.8e-05</td></tr>\n",
       "\t<tr><td>1-59343-C-T </td><td>2.3e-05</td></tr>\n",
       "\t<tr><td>1-62794-T-A </td><td>2.3e-05</td></tr>\n",
       "\t<tr><td>1-64927-G-T </td><td>8e-06  </td></tr>\n",
       "\t<tr><td>1-66269-A-T </td><td>0      </td></tr>\n",
       "\t<tr><td>1-66356-T-A </td><td>8.8e-05</td></tr>\n",
       "\t<tr><td>1-70991-C-T </td><td>0      </td></tr>\n",
       "\t<tr><td>1-72456-G-A </td><td>0      </td></tr>\n",
       "\t<tr><td>1-72588-A-G </td><td>0      </td></tr>\n",
       "\t<tr><td>1-73444-T-C </td><td>1e-05  </td></tr>\n",
       "\t<tr><td>1-75442-A-C </td><td>0      </td></tr>\n",
       "\t<tr><td>1-75852-C-T </td><td>8.5e-05</td></tr>\n",
       "\t<tr><td>1-82249-A-G </td><td>9.5e-05</td></tr>\n",
       "\t<tr><td>1-83962-A-AT</td><td>8.7e-05</td></tr>\n",
       "\t<tr><td>1-83996-A-T </td><td>0      </td></tr>\n",
       "\t<tr><td>1-96592-T-C </td><td>9e-06  </td></tr>\n",
       "\t<tr><td>1-100053-G-A</td><td>0      </td></tr>\n",
       "\t<tr><td>1-102596-G-A</td><td>2.1e-05</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 20 × 2 of type chr\n",
       "\\begin{tabular}{ll}\n",
       " variant\\_id & gnomAD\\_allele\\_freq\\\\\n",
       "\\hline\n",
       "\t 1-28198-A-T  & 0      \\\\\n",
       "\t 1-46402-C-CT & 0      \\\\\n",
       "\t 1-55119-C-G  & 2.8e-05\\\\\n",
       "\t 1-59343-C-T  & 2.3e-05\\\\\n",
       "\t 1-62794-T-A  & 2.3e-05\\\\\n",
       "\t 1-64927-G-T  & 8e-06  \\\\\n",
       "\t 1-66269-A-T  & 0      \\\\\n",
       "\t 1-66356-T-A  & 8.8e-05\\\\\n",
       "\t 1-70991-C-T  & 0      \\\\\n",
       "\t 1-72456-G-A  & 0      \\\\\n",
       "\t 1-72588-A-G  & 0      \\\\\n",
       "\t 1-73444-T-C  & 1e-05  \\\\\n",
       "\t 1-75442-A-C  & 0      \\\\\n",
       "\t 1-75852-C-T  & 8.5e-05\\\\\n",
       "\t 1-82249-A-G  & 9.5e-05\\\\\n",
       "\t 1-83962-A-AT & 8.7e-05\\\\\n",
       "\t 1-83996-A-T  & 0      \\\\\n",
       "\t 1-96592-T-C  & 9e-06  \\\\\n",
       "\t 1-100053-G-A & 0      \\\\\n",
       "\t 1-102596-G-A & 2.1e-05\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 20 × 2 of type chr\n",
       "\n",
       "| variant_id | gnomAD_allele_freq |\n",
       "|---|---|\n",
       "| 1-28198-A-T  | 0       |\n",
       "| 1-46402-C-CT | 0       |\n",
       "| 1-55119-C-G  | 2.8e-05 |\n",
       "| 1-59343-C-T  | 2.3e-05 |\n",
       "| 1-62794-T-A  | 2.3e-05 |\n",
       "| 1-64927-G-T  | 8e-06   |\n",
       "| 1-66269-A-T  | 0       |\n",
       "| 1-66356-T-A  | 8.8e-05 |\n",
       "| 1-70991-C-T  | 0       |\n",
       "| 1-72456-G-A  | 0       |\n",
       "| 1-72588-A-G  | 0       |\n",
       "| 1-73444-T-C  | 1e-05   |\n",
       "| 1-75442-A-C  | 0       |\n",
       "| 1-75852-C-T  | 8.5e-05 |\n",
       "| 1-82249-A-G  | 9.5e-05 |\n",
       "| 1-83962-A-AT | 8.7e-05 |\n",
       "| 1-83996-A-T  | 0       |\n",
       "| 1-96592-T-C  | 9e-06   |\n",
       "| 1-100053-G-A | 0       |\n",
       "| 1-102596-G-A | 2.1e-05 |\n",
       "\n"
      ],
      "text/plain": [
       "      variant_id   gnomAD_allele_freq\n",
       " [1,] 1-28198-A-T  0                 \n",
       " [2,] 1-46402-C-CT 0                 \n",
       " [3,] 1-55119-C-G  2.8e-05           \n",
       " [4,] 1-59343-C-T  2.3e-05           \n",
       " [5,] 1-62794-T-A  2.3e-05           \n",
       " [6,] 1-64927-G-T  8e-06             \n",
       " [7,] 1-66269-A-T  0                 \n",
       " [8,] 1-66356-T-A  8.8e-05           \n",
       " [9,] 1-70991-C-T  0                 \n",
       "[10,] 1-72456-G-A  0                 \n",
       "[11,] 1-72588-A-G  0                 \n",
       "[12,] 1-73444-T-C  1e-05             \n",
       "[13,] 1-75442-A-C  0                 \n",
       "[14,] 1-75852-C-T  8.5e-05           \n",
       "[15,] 1-82249-A-G  9.5e-05           \n",
       "[16,] 1-83962-A-AT 8.7e-05           \n",
       "[17,] 1-83996-A-T  0                 \n",
       "[18,] 1-96592-T-C  9e-06             \n",
       "[19,] 1-100053-G-A 0                 \n",
       "[20,] 1-102596-G-A 2.1e-05           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "library(jsonlite)\n",
    "\n",
    "freq_threshold = 0.0001\n",
    "ids = c()\n",
    "freqs = c()\n",
    "\n",
    "for (position in positions) {\n",
    "    position_data = fromJSON(position)\n",
    "    for (i in 1:dim(position_data$variants)[1]) {\n",
    "        vid = position_data$variants$vid[i]\n",
    "        freq = position_data$variants$gnomad$allAf[i]\n",
    "        if ( !is.null(freq) && !is.na(freq) && freq < freq_threshold ) {\n",
    "            ids = c(ids, vid)\n",
    "            freqs = c(freqs, freq)\n",
    "        }\n",
    "    }\n",
    "}\n",
    "\n",
    "freq_df = cbind(ids, freqs)\n",
    "colnames(freq_df) = c(\"variant_id\", \"gnomAD_allele_freq\")\n",
    "print(dim(freq_df))\n",
    "freq_df[1:20,]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fba1420c-35d6-43d1-9e26-6a425cf765f7",
   "metadata": {},
   "source": [
    "## Retrieve all relevant genes and their OMIM gene names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3c6400db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 27 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>gene</th><th scope=col>OMIM_gene_name</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;NULL&gt;</th><th scope=col>&lt;NULL&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;NULL&gt;</th><th scope=col>&lt;NULL&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ACAP3</td><td>NULL</td></tr>\n",
       "\t<tr><td>AGRN</td><td>Agrin</td></tr>\n",
       "\t<tr><td>AL627309.1</td><td>NULL</td></tr>\n",
       "\t<tr><td>AL645608.1</td><td>NULL</td></tr>\n",
       "\t<tr><td>B3GALT6</td><td>UDP-Gal:beta-Gal beta-1,3-galactosyltransferase polypeptide 6</td></tr>\n",
       "\t<tr><td>C1QTNF12</td><td>Family with sequence similarity 132, member A</td></tr>\n",
       "\t<tr><td>C1orf159</td><td>NULL</td></tr>\n",
       "\t<tr><td>CPTP</td><td>Glycolipid transfer protein domain-containing protein 1</td></tr>\n",
       "\t<tr><td>DVL1</td><td>Dishevelled segment polarity protein 1</td></tr>\n",
       "\t<tr><td>HES4</td><td>Hes family bHLH transcription factor 4</td></tr>\n",
       "\t<tr><td>INTS11</td><td>Cleavage and polyadenylation-specific factor 3-like</td></tr>\n",
       "\t<tr><td>ISG15</td><td>ISG15 ubiquitin-like modifier</td></tr>\n",
       "\t<tr><td>KLHL17</td><td>Kelch-like 17</td></tr>\n",
       "\t<tr><td>NOC2L</td><td>NOC2-like nucleolar-associated transcriptional repressor</td></tr>\n",
       "\t<tr><td>OR4F5</td><td>NULL</td></tr>\n",
       "\t<tr><td>PERM1</td><td>PPARGC1-and ESRR-induced regulator, muscle, 1</td></tr>\n",
       "\t<tr><td>PLEKHN1</td><td>NULL</td></tr>\n",
       "\t<tr><td>PUSL1</td><td>NULL</td></tr>\n",
       "\t<tr><td>RNF223</td><td>NULL</td></tr>\n",
       "\t<tr><td>SAMD11</td><td>Sterile alpha motif domain-containing protein 11</td></tr>\n",
       "\t<tr><td>SCNN1D</td><td>Sodium channel epithelial 1, subunit delta</td></tr>\n",
       "\t<tr><td>SDF4</td><td>Stromal cell-derived factor 4</td></tr>\n",
       "\t<tr><td>TAS1R3</td><td>Taste receptor type 1, member 3</td></tr>\n",
       "\t<tr><td>TNFRSF18</td><td>Tumor necrosis factor receptor superfamily, member 18</td></tr>\n",
       "\t<tr><td>TNFRSF4</td><td>Tumor necrosis factor receptor superfamily, member 4</td></tr>\n",
       "\t<tr><td>TTLL10</td><td>NULL</td></tr>\n",
       "\t<tr><td>UBE2J2</td><td>NULL</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 27 × 2\n",
       "\\begin{tabular}{ll}\n",
       " gene & OMIM\\_gene\\_name\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & ⋯ & <NULL> & <NULL> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <NULL> & <NULL>\\\\\n",
       "\\hline\n",
       "\t ACAP3 & NULL\\\\\n",
       "\t AGRN & Agrin\\\\\n",
       "\t AL627309.1 & NULL\\\\\n",
       "\t AL645608.1 & NULL\\\\\n",
       "\t B3GALT6 & UDP-Gal:beta-Gal beta-1,3-galactosyltransferase polypeptide 6\\\\\n",
       "\t C1QTNF12 & Family with sequence similarity 132, member A\\\\\n",
       "\t C1orf159 & NULL\\\\\n",
       "\t CPTP & Glycolipid transfer protein domain-containing protein 1\\\\\n",
       "\t DVL1 & Dishevelled segment polarity protein 1\\\\\n",
       "\t HES4 & Hes family bHLH transcription factor 4\\\\\n",
       "\t INTS11 & Cleavage and polyadenylation-specific factor 3-like\\\\\n",
       "\t ISG15 & ISG15 ubiquitin-like modifier\\\\\n",
       "\t KLHL17 & Kelch-like 17\\\\\n",
       "\t NOC2L & NOC2-like nucleolar-associated transcriptional repressor\\\\\n",
       "\t OR4F5 & NULL\\\\\n",
       "\t PERM1 & PPARGC1-and ESRR-induced regulator, muscle, 1\\\\\n",
       "\t PLEKHN1 & NULL\\\\\n",
       "\t PUSL1 & NULL\\\\\n",
       "\t RNF223 & NULL\\\\\n",
       "\t SAMD11 & Sterile alpha motif domain-containing protein 11\\\\\n",
       "\t SCNN1D & Sodium channel epithelial 1, subunit delta\\\\\n",
       "\t SDF4 & Stromal cell-derived factor 4\\\\\n",
       "\t TAS1R3 & Taste receptor type 1, member 3\\\\\n",
       "\t TNFRSF18 & Tumor necrosis factor receptor superfamily, member 18\\\\\n",
       "\t TNFRSF4 & Tumor necrosis factor receptor superfamily, member 4\\\\\n",
       "\t TTLL10 & NULL\\\\\n",
       "\t UBE2J2 & NULL\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 27 × 2\n",
       "\n",
       "| gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene ⋯ | OMIM_gene_name &lt;NULL&gt; | gene &lt;NULL&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;chr&gt; | gene &lt;chr&gt; | OMIM_gene_name &lt;NULL&gt; | gene &lt;NULL&gt; |\n",
       "|---|---|\n",
       "| ACAP3 | NULL |\n",
       "| AGRN | Agrin |\n",
       "| AL627309.1 | NULL |\n",
       "| AL645608.1 | NULL |\n",
       "| B3GALT6 | UDP-Gal:beta-Gal beta-1,3-galactosyltransferase polypeptide 6 |\n",
       "| C1QTNF12 | Family with sequence similarity 132, member A |\n",
       "| C1orf159 | NULL |\n",
       "| CPTP | Glycolipid transfer protein domain-containing protein 1 |\n",
       "| DVL1 | Dishevelled segment polarity protein 1 |\n",
       "| HES4 | Hes family bHLH transcription factor 4 |\n",
       "| INTS11 | Cleavage and polyadenylation-specific factor 3-like |\n",
       "| ISG15 | ISG15 ubiquitin-like modifier |\n",
       "| KLHL17 | Kelch-like 17 |\n",
       "| NOC2L | NOC2-like nucleolar-associated transcriptional repressor |\n",
       "| OR4F5 | NULL |\n",
       "| PERM1 | PPARGC1-and ESRR-induced regulator, muscle, 1 |\n",
       "| PLEKHN1 | NULL |\n",
       "| PUSL1 | NULL |\n",
       "| RNF223 | NULL |\n",
       "| SAMD11 | Sterile alpha motif domain-containing protein 11 |\n",
       "| SCNN1D | Sodium channel epithelial 1, subunit delta |\n",
       "| SDF4 | Stromal cell-derived factor 4 |\n",
       "| TAS1R3 | Taste receptor type 1, member 3 |\n",
       "| TNFRSF18 | Tumor necrosis factor receptor superfamily, member 18 |\n",
       "| TNFRSF4 | Tumor necrosis factor receptor superfamily, member 4 |\n",
       "| TTLL10 | NULL |\n",
       "| UBE2J2 | NULL |\n",
       "\n"
      ],
      "text/plain": [
       "      gene       OMIM_gene_name                                               \n",
       " [1,] ACAP3      NULL                                                         \n",
       " [2,] AGRN       Agrin                                                        \n",
       " [3,] AL627309.1 NULL                                                         \n",
       " [4,] AL645608.1 NULL                                                         \n",
       " [5,] B3GALT6    UDP-Gal:beta-Gal beta-1,3-galactosyltransferase polypeptide 6\n",
       " [6,] C1QTNF12   Family with sequence similarity 132, member A                \n",
       " [7,] C1orf159   NULL                                                         \n",
       " [8,] CPTP       Glycolipid transfer protein domain-containing protein 1      \n",
       " [9,] DVL1       Dishevelled segment polarity protein 1                       \n",
       "[10,] HES4       Hes family bHLH transcription factor 4                       \n",
       "[11,] INTS11     Cleavage and polyadenylation-specific factor 3-like          \n",
       "[12,] ISG15      ISG15 ubiquitin-like modifier                                \n",
       "[13,] KLHL17     Kelch-like 17                                                \n",
       "[14,] NOC2L      NOC2-like nucleolar-associated transcriptional repressor     \n",
       "[15,] OR4F5      NULL                                                         \n",
       "[16,] PERM1      PPARGC1-and ESRR-induced regulator, muscle, 1                \n",
       "[17,] PLEKHN1    NULL                                                         \n",
       "[18,] PUSL1      NULL                                                         \n",
       "[19,] RNF223     NULL                                                         \n",
       "[20,] SAMD11     Sterile alpha motif domain-containing protein 11             \n",
       "[21,] SCNN1D     Sodium channel epithelial 1, subunit delta                   \n",
       "[22,] SDF4       Stromal cell-derived factor 4                                \n",
       "[23,] TAS1R3     Taste receptor type 1, member 3                              \n",
       "[24,] TNFRSF18   Tumor necrosis factor receptor superfamily, member 18        \n",
       "[25,] TNFRSF4    Tumor necrosis factor receptor superfamily, member 4         \n",
       "[26,] TTLL10     NULL                                                         \n",
       "[27,] UBE2J2     NULL                                                         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gene_symbols = list()\n",
    "omim_gene_names = list()\n",
    "\n",
    "for (gene in genes) {\n",
    "    gene_data = fromJSON(gene)\n",
    "    gene_symbols = append(gene_symbols, gene_data$name)\n",
    "    omim_gene_name = gene_data$omim$geneName[1]\n",
    "    if ( is.null(omim_gene_name) ) {\n",
    "        omim_gene_names = append(omim_gene_names, list(NULL))\n",
    "    }\n",
    "    else {\n",
    "        omim_gene_names = append(omim_gene_names, omim_gene_name)\n",
    "    }\n",
    "}\n",
    "\n",
    "gene_df = cbind(gene_symbols, omim_gene_names)\n",
    "colnames(gene_df) = c(\"gene\", \"OMIM_gene_name\")\n",
    "gene_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
